### PR TITLE
export debug tools

### PR DESCRIPTION
**Pull request by kevinushey**
So that these symbols can be called from a debugger, even in package builds. We need to do some extra work for package builds on macOS to:

- Make sure we only export symbols that we need to be available in the final executable;
- Ensure that `strip` only removes debug symbols and not the "local but referenced by global" symbols we use above.

Also includes some CMake hygiene changes; mainly so that editors can infer the appropriate syntax highlight rules for these files without any trouble.

### CODE REVIEW

File: cmake/globals.cmake [hunk 1]
```diff
@@ -435,15 +433,18 @@ macro(add_stripped_executable _target)
...
-            set(STRIP_TARGET "${_target}.app/contents/MacOS/${_target}")
+            set(STRIP_TARGET "${_target}.app/Contents/MacOS/${_target}")
          else()
             set(STRIP_TARGET "${_target}")
          endif()
+         
          add_custom_command(TARGET ${_target} POST_BUILD
                             COMMAND dsymutil -o ./${_target}.dSYM ${STRIP_TARGET}
-                            COMMAND strip ${STRIP_TARGET}
+                            COMMAND strip -x -S ${STRIP_TARGET}
```
Comment (kevinushey) [inline cmake/globals.cmake:445]: For reference, from `man strip`:

       -S     Remove the debugging symbol table entries (those created by the -g option to cc(1) and
              other compilers).

       -x     Remove all local symbols (saving only global symbols).

And

       If the executable loads objects, however, the global symbols that the objects reference from the
       executable also must not be stripped.  In this case, when linking the executable you should use
       the `-exported_symbols_list` option of the link editor ld(1) to limit which symbols can be
       referenced by the objects.  Then you only need to strip local and debug symbols, like that:
              % strip -x -S executable

I've tried to follow that advice here and elsewhere in the PR.

File: cmake/init.cmake [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#
+# init.cmake
+#
+# Copyright (C) 2022 by Posit Software, PBC
```
Comment (gtritchie) [inline cmake/init.cmake:4]: 2024

File: src/cpp/r/CMakeLists.txt [hunk 3]
```diff
@@ -100,7 +100,7 @@ target_link_libraries(rstudio-r rstudio-core)
 if(APPLE)
    target_link_libraries(rstudio-r "-undefined dynamic_lookup")
 else()
-   target_link_libraries(rstudio-r "${LIBR_LIBRARIES}")
+   target_link_libraries(rstudio-r ${LIBR_LIBRARIES} ${YAML_CPP_LIBRARIES})
```
Comment (gtritchie) [inline src/cpp/r/CMakeLists.txt:103]: Why was YAML_CPP_LIBRARIES added here?

**Review by gtritchie — APPROVED** [review]
LGTM! You'll probably want to handle merging open-source into pro in case it requires any tricky merging.

File: src/cpp/r/CMakeLists.txt (hunk 3 above)
Comment (kevinushey) [reply src/cpp/r/CMakeLists.txt:103]: The Linux build was complaining about a lack of yaml-cpp; my only guess is that this is somehow related to the version of CMake I was using, but it shouldn't hurt at least...

**Pull request merged**